src/msg/simple/Pipe.cc

```diff
@@ -2035,8 +2034,9 @@ int Pipe::read_message(Message **pm, AuthSessionHandler* auth_handler)
     if (msgr->crcflags & MSG_CRC_HEADER) {
       footer.front_crc = old_footer.front_crc;
       footer.middle_crc = old_footer.middle_crc;
-      footer.data_crc = old_footer.data_crc;
     }
+    if (msgr->crcflags & MSG_CRC_DATA)
+      footer.data_crc = old_footer.data_crc;
     footer.sig = 0;
```

Comment:
i think this block may as well unconditionally copy these fields.. we're just translating the footer formats.  the checks some later and we may ignore them but that's fine